### ISSUE TITLE
libdc1394 error: Failed to initialize libdc1394 when ``import cv2`` in python code

### ISSUE THREAD

**Issue by wkentaro** [jsk-ros-pkg/jsk_travis#187]
a question at StackOverflow: http://stackoverflow.com/questions/12689304/ctypes-error-libdc1394-error-failed-to-initialize-libdc1394

Maybe this is caused by opencv.

Also it raises `ImportError` for numpy.core.multiarray.
```
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] ImportError: numpy.core.multiarray failed to import
[jsk_perception] Traceback (most recent call last):
[jsk_perception]   File "/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/node_scripts/image_publisher.py", line 4, in <module>
[jsk_perception]     import cv2
[jsk_perception] Traceback (most recent call last):
```

Maybe something is different in jenkins from standard ubuntu environment?
(I have no error when `import cv2` on my PC)

**Comment by furushchev** [comment]
You can ignore this error message.
it is just warning.

**Comment by furushchev** [comment]
I checked

```
docker run -it ubuntu:trustty /bin/bash
# apt-get update
# apt-get install python-opencv
# python
>> try:
...     import cv2
... except Exception as e:
...     print "hogehoge", e
...
libdc1394 error: Failed to initialize libdc1394
>> # does not show "hogehoge"
```

**Comment by wkentaro** [comment]
It seems that this warning does not raise only when importing cv2.

```
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] Traceback (most recent call last):
[jsk_perception]   File "/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/node_scripts/mask_image_to_label.py", line 6, in <module>
[jsk_perception]     from jsk_topic_tools import ConnectionBasedTransport
[jsk_perception]   File "/workspace/ros/ws_jsk_recognition/devel/lib/python2.7/dist-packages/jsk_topic_tools/__init__.py", line 35, in <module>
[jsk_perception]     exec(__fh.read())
[jsk_perception]   File "<string>", line 4, in <module>
[jsk_perception]   File "/workspace/ros/ws_jsk_recognition/src/jsk-ros-pkg/jsk_common/jsk_topic_tools/src/jsk_topic_tools/transport.py", line 6, in <module>
[jsk_perception]     import rospy
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/__init__.py", line 47, in <module>
[jsk_perception] Traceback (most recent call last):
[jsk_perception]   File "/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/node_scripts/image_publisher.py", line 6, in <module>
[jsk_perception]     import rospy
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/__init__.py", line 47, in <module>
[jsk_perception] Traceback (most recent call last):
[jsk_perception]   File "/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/node_scripts/image_publisher.py", line 6, in <module>
[jsk_perception]     import rospy
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/__init__.py", line 49, in <module>
[jsk_perception]         from std_msgs.msg import Header
[jsk_perception]     from std_msgs.msg import Header
[jsk_perception] from .client import spin, myargv, init_node, \
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/std_msgs/msg/__init__.py", line 1, in <module>
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/std_msgs/msg/__init__.py", line 1, in <module>
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/client.py", line 49, in <module>
[jsk_perception]     import rosgraph
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rosgraph/__init__.py", line 36, in <module>
[jsk_perception]         from ._Bool import *
[jsk_perception] from ._Bool import *
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/std_msgs/msg/_Bool.py", line 4, in <module>
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/std_msgs/msg/_Bool.py", line 2, in <module>
[jsk_perception]     from . masterapi import Master, MasterFailure, MasterError, MasterException
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rosgraph/masterapi.py", line 41, in <module>
[jsk_perception]     from xmlrpc.client import ServerProxy  # Python 3.x
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rosgraph/xmlrpc.py", line 59, in <module>
[jsk_perception]     from SimpleXMLRPCServer import SimpleXMLRPCServer #Python 2.x
[jsk_perception]   File "/usr/lib/python2.7/SimpleXMLRPCServer.py", line 102, in <module>
[jsk_perception]     import genpy
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/genpy/__init__.py", line 34, in <module>
[jsk_perception]     import sys
[jsk_perception] KeyboardInterrupt
[jsk_perception]     import xmlrpclib
[jsk_perception]   File "/usr/lib/python2.7/xmlrpclib.py", line 145, in <module>
[jsk_perception]     import httplib
[jsk_perception]   File "/usr/lib/python2.7/httplib.py", line 73, in <module>
[jsk_perception]     from urlparse import urlsplit
[jsk_perception]   File "/usr/lib/python2.7/urlparse.py", line 333, in <module>
[jsk_perception]     for a in _hexdig for b in _hexdig)
[jsk_perception]   File "/usr/lib/python2.7/urlparse.py", line 333, in <genexpr>
[jsk_perception]     for a in _hexdig for b in _hexdig)
[jsk_perception] KeyboardInterrupt
[jsk_perception]     from . message import Message, SerializationError, DeserializationError, MessageException, struct_I
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/genpy/message.py", line 45, in <module>
[jsk_perception]     import genmsg
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/genmsg/__init__.py", line 34, in <module>
[jsk_perception]     from . gentools import compute_md5, compute_full_text, compute_md5_text
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/genmsg/gentools.py", line 54, in <module>
[jsk_perception]     from .msg_loader import load_depends
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/genmsg/msg_loader.py", line 55, in <module>
[jsk_perception]     from . srvs import SrvSpec
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/genmsg/srvs.py", line 36, in <module>
[jsk_perception]     """
[jsk_perception] KeyboardInterrupt
[jsk_perception] libdc1394 error: Failed to initialize libdc1394
[jsk_perception] testtest_stdout ... FAILURE!
[jsk_perception] FAILURE: Test Fixture Nodes ['label_image_delayed'] failed to launch
[jsk_perception]   File "/usr/lib/python2.7/unittest/case.py", line 331, in run
[jsk_perception]     testMethod()
[jsk_perception]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/runner.py", line 121, in fn
[jsk_perception]     self.assert_(not failed, "Test Fixture Nodes %s failed to launch"%failed)
[jsk_perception]   File "/usr/lib/python2.7/unittest/case.py", line 424, in assertTrue
[jsk_perception]     raise self.failureException(msg)
```

It raises at every time for rostest.


I leave this issue for a while until I found the solution.

**Comment by garaemon** [comment]
http://stackoverflow.com/questions/18491641/opencv-and-numpy-interacting-badly

> It seems that for some reason, loading the numpy and cv libraries, in that order, constrains unpack to use 32 and 64 bit precision and return 0 for lower values.

**Comment by garaemon** [comment]
@wkentaro Have you tried below?

```
sudo ln /dev/null /dev/raw1394
```

**Comment by wkentaro** [comment]
No, need to run it in travis.sh? to fix this problem on jekins?

**Comment by garaemon** [comment]
maybe travis_jenkins.py

**Comment by wkentaro** [comment]
Ok I will try when I encounter this next time.

**Comment by garaemon** [comment]
Disappeared?

**Comment by wkentaro** [comment]
Not yet released jsk_travis after the PR #217.
Could you please release 0.2.3?

**Comment by wkentaro** [comment]
Seems disappeared. So closing this.
